src/lightning_app/utilities/proxies.py

```diff
@@ -123,6 +123,7 @@ def __call__(self, *args, **kwargs):
                     # overriding the return value for next loop iteration.
                     _send_data_to_caller_queue(self.work, self.caller_queue, data, call_hash)
         if not self.parallel:
+            # Question: Didn't understand this exception. Can you improve it?
```

Comment:
Here is how it works.

When running a work with `parallel=False`, we can't simply continue the flow execution and therefore when the work hasn't completed.
In order to enable this, we raise an exception and re-run the flow from the beginning.